src/SOS/Strike/clrma/clrma.cpp

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "exts.h"
+#include "managedanalysis.h"
+
+HRESULT CLRMACreateInstance(ICLRManagedAnalysis** ppCLRMA);
+HRESULT CLRMAReleaseInstance();
+
+ICLRManagedAnalysis* g_managedAnalysis = nullptr;
+
+int g_clrmaGlobalFlags = ClrmaGlobalFlags::LoggingEnabled | ClrmaGlobalFlags::DacClrmaEnabled | ClrmaGlobalFlags::ManagedClrmaEnabled;
+
+//
+// Exports
+//
+
+HRESULT
+CLRMACreateInstance(ICLRManagedAnalysis** ppCLRMA)
+{
+    HRESULT hr = E_UNEXPECTED;
+
+    if (ppCLRMA == nullptr)
+    {
+        return E_INVALIDARG;
+    }
+
+    *ppCLRMA = nullptr;
+
+    if (g_managedAnalysis == nullptr)
+    {
+        g_managedAnalysis = new (std::nothrow) ClrmaManagedAnalysis();
+        if (g_managedAnalysis == nullptr)
+        {
+           return E_OUTOFMEMORY;
+        }
+        OnUnloadTask::Register(([]() { CLRMAReleaseInstance(); }));
+    }
+
+    g_managedAnalysis->AddRef();
+    *ppCLRMA = g_managedAnalysis;
+    return S_OK;
+}
+
+HRESULT
+CLRMAReleaseInstance()
+{
+    TraceInformation("CLRMAReleaseInstance\n");
+    if (g_managedAnalysis != nullptr)
+    {
+        g_managedAnalysis->Release();
+        g_managedAnalysis = nullptr;
+    }
+    return S_OK;
+}
+
+DECLARE_API(clrmaconfig)
+{
+    INIT_API_EXT();
+
+    BOOL bEnable = FALSE;
+    BOOL bDisable = FALSE;
+    BOOL bDacClrma = FALSE;
+    BOOL bManagedClrma = FALSE;
+    BOOL bLogging = FALSE;
+
+    CMDOption option[] =
+    {   // name, vptr, type, hasValue
+        {"-enable", &bEnable, COBOOL, FALSE},
+        {"-disable", &bDisable, COBOOL, FALSE},
+        {"-dac", &bDacClrma, COBOOL, FALSE},
+        {"-managed", &bManagedClrma, COBOOL, FALSE},
+        {"-logging", &bLogging, COBOOL, FALSE},
+    };
+
+    if (!GetCMDOption(args, option, ARRAY_SIZE(option), NULL, 0, NULL))
+    {
+        return E_INVALIDARG;
+    }
+
+    if (bEnable)
+    {
+        if (bDacClrma)
+        {
+            g_clrmaGlobalFlags |= ClrmaGlobalFlags::DacClrmaEnabled;
+        }
+        if (bManagedClrma)
+        {
+            g_clrmaGlobalFlags |= ClrmaGlobalFlags::ManagedClrmaEnabled;
+        }
+        if (bLogging)
+        {
+            g_clrmaGlobalFlags |= ClrmaGlobalFlags::LoggingEnabled;
+        }
+    }
+    else if (bDisable)
+    {
+        if (bDacClrma)
+        {
+            g_clrmaGlobalFlags &= ~ClrmaGlobalFlags::DacClrmaEnabled;
+        }
+        if (bManagedClrma)
+        {
+            g_clrmaGlobalFlags &= ~ClrmaGlobalFlags::ManagedClrmaEnabled;
+        }
+        if (bLogging)
+        {
+            g_clrmaGlobalFlags &= ~ClrmaGlobalFlags::LoggingEnabled;
+        }
+    }
+
+    ExtOut("CLRMA logging:              %s\n", g_clrmaGlobalFlags & ClrmaGlobalFlags::LoggingEnabled ? "enabled (disable with '-disable -logging')" : "disabled (enable with '-enable -logging')");
```

Comment:
Should the bitwise `&` be in parenthesis due to operator precedence with the ternary conditional?